GILREICH1/grassHeap
# Feat/auth

## BIG PR!
### GrassHeap 2.0 ?

**For the PR that brings you Auth, a lot has changed:**
- [x] New userContxt to access properties of the User in Front-End.
- [x] FE Tests added.
- [x] New "User" collection added to DB to store User info.
- [x] User created tasks and 'myPlants' come from user endpoint, saved to User document in DB.

**To Test**
- Register for Auth0 account and create 1 SPA application and 1 machine to machine appliction (<a href="https://auth0.com/blog/complete-guide-to-react-user-authentication" target="_blank">This tutorial</a> might be helpful) 
- pull this branch, run `npm i`
- Run server and client locally and ensure client is making requests to local server.

**Next Steps?** 
- [ ] Some state management might be useful (React Query? )
- [ ]  more tests + hook to run them?
- [ ] solve linter dbl quotes problems
